cube/cube.cpp

```diff
@@ -2156,136 +1963,124 @@ void Demo::prepare_render_pass() {
             .setDependencyFlags(vk::DependencyFlags()),
     };
 
-    auto const rp_info = vk::RenderPassCreateInfo()
-                             .setAttachmentCount(2)
-                             .setPAttachments(attachments)
-                             .setSubpassCount(1)
-                             .setPSubpasses(&subpass)
-                             .setDependencyCount(2)
-                             .setPDependencies(dependencies);
-
-    auto result = device.createRenderPass(&rp_info, nullptr, &render_pass);
-    VERIFY(result == vk::Result::eSuccess);
+    const auto render_pass_result = device.createRenderPass(
+        vk::RenderPassCreateInfo().setAttachments(attachments).setSubpasses(subpass).setDependencies(dependencies));
+    VERIFY(render_pass_result.result == vk::Result::eSuccess);
+    render_pass = render_pass_result.value;
 }
 
 vk::ShaderModule Demo::prepare_shader_module(const uint32_t *code, size_t size) {
-    const auto moduleCreateInfo = vk::ShaderModuleCreateInfo().setCodeSize(size).setPCode(code);
+    const auto shader_module_return = device.createShaderModule(vk::ShaderModuleCreateInfo().setCodeSize(size).setPCode(code));
+    VERIFY(shader_module_return.result == vk::Result::eSuccess);
 
-    vk::ShaderModule module;
-    auto result = device.createShaderModule(&moduleCreateInfo, nullptr, &module);
-    VERIFY(result == vk::Result::eSuccess);
-
-    return module;
+    return shader_module_return.value;
 }
 
-void Demo::prepare_texture_buffer(const char *filename, texture_object *tex_obj) {
+void Demo::prepare_texture_buffer(const char *filename, texture_object &tex_obj) {
+    vk::SubresourceLayout tex_layout;
     int32_t tex_width;
     int32_t tex_height;
-
-    if (!loadTexture(filename, NULL, NULL, &tex_width, &tex_height)) {
+    if (!loadTexture(filename, NULL, tex_layout, tex_width, tex_height)) {
```

Comment:
`nullptr` instead of `NULL`

Reply:
Fixed. (and everywhere else I saw NULL.)